pkg/lib/oauth/handler/handler_authz.go

```diff
@@ -108,7 +125,18 @@ func (h *AuthorizationHandler) Handle(r protocol.AuthorizationRequest) httputil.
 			Response:     protocol.NewErrorResponse("unauthorized_client", "invalid client ID"),
 		}
 	}
-	redirectURI, errResp := parseRedirectURI(client, h.HTTPProto, h.HTTPOrigin, h.AppDomains, r)
+
+	oauthSessionEntry := oauthsession.NewEntry(oauthsession.T{
+		AuthorizationRequest: r,
+	})
+	err := h.OAuthSessionService.Save(oauthSessionEntry)
+	if err != nil {
+		return authorizationResultError{
+			ResponseMode: r.ResponseMode(),
+			Response:     protocol.NewErrorResponse("server_error", "internal server error"),
+		}
+	}
+	redirectURI, errResp := parseAuthzRedirectURI(client, h.UIURLBuilder, h.HTTPProto, h.HTTPOrigin, h.AppDomains, oauthSessionEntry, r)
```

Comment:
Please do not alter the flow. h.OAuthSessionService.Save was originally used in doHandle. Now it was used more than once.